src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
+            if (arguments.isEmpty()) {
+                throw noArgumentValueWasCaptured();
+            }
 
+            return (T) arguments.get(arguments.size() - 1);
+        }
     }
 
     public List<T> getAllValues() {
-        return (List) arguments;
+        return Arrays.asList((T[]) arguments.toArray());
```

Comment:
@ChristianSchwarz Sorry, I didn't understand your original comment correctly. Yes, I agree, it should be forbidden to change the contents of this collection and returning immutable collection here is the right thing to do. In fact, I'm already doing so - `Arrays.asList()` factory creates immutable lists. You can't alter the contents of a list produced by this factory.